File: pracaInż/Models/Entities/CompanyStructure/Factory.cs
using pracaInż.Models.DTO.Factories;

namespace pracaInż.Models.Entities.CompanyStructure
{
    public class Factory
    {
        public int Id { get; set; }
        public string City { get; set; }
        public string Street { get; set; }
        public int StreetNumber { get; set; }
        public string PostalCode { get; set; }

        public List<Department> Departments { get; set; }

        public Factory()
        {

        }

        public Factory(AddFactoryDTO factory)
        {
            City = factory.City;
            Street = factory.Street;
            StreetNumber = factory.StreetNumber;
            PostalCode = factory.PostalCode;
        }
        
    }
}

File: pracaInż/Frotnend/src/app/Models/Factory.ts
export interface Factory {
    id: number;
    city: string;
    street: string;
    streetNumber: number;
    postalCode: string;
}

export interface SelectFactoryDTO{
    id: number;
    city: string;
}

export interface AddFactoryDTO{
    city: string;
    Street: string;
    streetNumber: number;
    postalCode: string;  
}